sql/wsrep_mysqld.cc

```diff
@@ -3168,7 +3168,8 @@ void wsrep_handle_mdl_conflict(MDL_context *requestor_ctx,
       }
     }
     else if (granted_thd->lex->sql_command == SQLCOM_FLUSH ||
-             granted_thd->mdl_context.has_explicit_locks())
+            (granted_thd->system_thread != NON_SYSTEM_THREAD &&
+             granted_thd->mdl_context.has_explicit_locks()))
```

Comment:
Please add comment here and modify below WSREP_DEBUG output to correctly state case when this is not FLUSH.

Reply:
Sure, I will print the statement itself.